.github/workflows/public-api.yml

```diff
@@ -0,0 +1,71 @@
+name: Public API
+on:
+    push:
+        branches:
+            - "*"
+    create:
+        tags:
+            - "*"
+    pull_request_target:
+permissions:
+    issues: write
+    pull-requests: write
+jobs:
+    Check:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v2
+              with:
+                  token: ${{ secrets.PUSHABLE_GITHUB_TOKEN }}
+                  ref: "refs/pull/${{ github.event.number }}/merge"
+                  path: inbound_pr
+            - name: Add workloads for restore
+              run: |
+                  dotnet workload install android
+            - name: Checkout submodules, configure git
+              run: |
+                  git -c submodule.third_party/git-hooks.update=none submodule update --init --recursive
+                  git config --local user.email "9011267+dotnet-bot@users.noreply.github.com"
+                  git config --local user.name "The Silk.NET Automaton" 
+            - name: Cache .tmp, ~/.nuget/packages
+              uses: actions/cache@v2
+              with:
+                  path: |
+                      .tmp
+                      ~/.nuget/packages
+                  key: ${{ runner.os }}-${{ hashFiles('**/global.json', '**/*.csproj') }}
```

Comment:
Why only if a csproj changes?

Reply:
This was originally NUKE-generated, but the only time the contents of ~/.nuget/packages will be tangibly different is if the PackageReferences and by extension csprojs change

Reply:
Oh I didn't see the path thing above